gc/base/standard/ParallelGlobalGC.cpp

```diff
@@ -478,7 +511,7 @@ MM_ParallelGlobalGC::mainThreadGarbageCollect(MM_EnvironmentBase *env, MM_Alloca
 			_collectionStatistics._tenureFragmentation |= MACRO_FRAGMENTATION;
 		}
 
-		mainThreadCompact(env, allocDescription, rebuildMarkBits);
+		mainThreadCompact(env, allocDescription, rebuildMarkBits || env->_cycleState->_gcCode.shouldClearHeap());
```

Comment:
also add a comment here
while normally mark map is not required for walking the heap after a  compact, the clearing callback asserts that everything that is not cleared is a marked object, so we need to rebuild a mark map for clearing